engine/classes/Elgg/Database.php

```diff
@@ -276,7 +276,7 @@ public function updateData($query, $get_num_rows = false, array $params = []) {
 		if ($get_num_rows) {
 			return $stmt->rowCount();
 		} else {
-			return true;
+			return (bool) $stmt->rowCount();
```

Comment:
This was a bad assumption. I will instead throw from database mock for non-select queries if no spec is defined, so that we don't miss any queries we should be mocking.